File: testng-core/src/test/java/test/hook/samples/HookSuccessWithListenerSample.java
package test.hook.samples;

import static org.assertj.core.api.Assertions.assertThat;
import static test.hook.HookableTest.*;

import org.testng.IHookCallBack;
import org.testng.IHookable;
import org.testng.ITestResult;
import org.testng.Reporter;
import org.testng.annotations.Listeners;
import org.testng.annotations.Test;

@Listeners(HookSuccessWithListenerSample.HookListener.class)
public class HookSuccessWithListenerSample {

  @Test
  public void verify() {
    ITestResult itr = Reporter.getCurrentTestResult();
    boolean attribute = Boolean.parseBoolean(itr.getAttribute(HOOK_INVOKED_ATTRIBUTE).toString());
    assertThat(attribute).isTrue();
  }

  public static class HookListener implements IHookable {

    @Override
    public void run(IHookCallBack callBack, ITestResult testResult) {
      testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true");
      callBack.runTestMethod(testResult);
    }
  }
}

File: testng-core/src/test/kotlin/test/hook/samples/HookSuccessWithListenerSample.kt
package test.hook.samples

import test.hook.*

import org.testng.Assert
import org.testng.IHookCallBack
import org.testng.IHookable
import org.testng.ITestResult
import org.testng.Reporter
import org.testng.annotations.Listeners
import org.testng.annotations.Test

@Listeners(HookSuccessWithListenerSample.HookListener::class)
class HookSuccessWithListenerSample {

    @Test
    fun verify() {
        val itr = Reporter.getCurrentTestResult()
        val attribute = itr.getAttribute(HOOK_INVOKED_ATTRIBUTE).toString().toBoolean()
        Assert.assertTrue(attribute)
    }

    class HookListener : IHookable {
        override fun run(callBack: IHookCallBack, testResult: ITestResult) {
            testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
            callBack.runTestMethod(testResult)
        }
    }
}
